.github/workflows/build.yaml

```diff
@@ -35,3 +38,58 @@ jobs:
           tags: "ghcr.io/${{ github.repository }}:${{ github.sha }}"
           cache-from: type=gha,scope=${{ github.repository }}
           cache-to: type=gha,scope=${{ github.repository }},mode=max
+
+  trivy-scan-code:
+    needs: build
+    name: Trivy Vulnerability Scan for Code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Aqua Security Trivy Scan
+        uses: aquasecurity/trivy-action@master
+        with:
+          scan-type: 'fs'
+          security-checks: 'vuln,secret,config'
+          ignore-unfixed: true
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: 'trivy-results.sarif'
+          category: 'code'
+
+      - name: Image Scan ghcr
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'ghcr.io/${{ github.repository }}:${{ github.sha }}'
+          format: 'sarif'
+          output: 'trivy-image-results-ghcr.sarif'
+          exit-code: '1'
```

Comment:
When does trivy exit with one? and why should we use it?

Reply:
With the flag exit-code: '1' if there is an error, the build with turn red on the PR.

Reply:
1. Double check the criteria, where is it supposed to fail? On this particular step? If so, shouldn't it be failing right now?

2. We already have a build failure that is set thanks to codeql. codeql have a database (for issues marked as won't fix and so on), therefore the whole context. And trivy on its own, will not be aware of this information. Therefore we should restrain this step from failing and leave this review to codeql.

The reason why we would like this step to fail, is if trivy was unable to perform the scan for whatever reason.

Let me know what you think with this logic.

Reply:
Ok I see, I don't necessarily agree with you. The idea is that the security processes are implemented thoroughly (even if it's a bit annoying for the user) and the fact that the build is red will put an additional alert that will unconsciously force the user to modify his code at least a bit before pushing